draft-proxied-svcb-headers.md

```diff
@@ -17,132 +17,545 @@ author:
+  as well as SVCB records for a name specified in this header.
+
+This document also defines two new response headers:
+
+* Proxy-DNS-SVCB: encodes SVCB records associated with requested name.
+* Proxy-DNS-Used: encodes DNS information used by the Proxy for making this connection.
+
+*TODO*: do we need a Terminology section, or a reference to Section 1.4 of {{SVCB}} as well as {{!RFC8499}}?
 
 ## Requirements
 
 {::boilerplate bcp14}
 
-# SVCB Request Header Field {#svcb-params-request}
 
-Clients can request SVCB parameters with the Structured Header
-{{!RFC8941}} "DNS-SVCB-Keys". Its value MUST
-be an sf-list whose members are sf-integer items that MUST NOT contain parameters. Its
-ABNF is:
+# Definitions of request and response header fields
+
+## Proxy-DNS-Request Request Header Field {#proxy-dns-request}
+
+Clients can send the Structured Header {{!RFC8941}} "Proxy-DNS-Request"
+to request that proxies send them DNS-related information.
+Its value MUST be an sf-item whose type is sf-string followed by a set of parameters.
+
+Its ABNF is:
 
 ~~~ abnf
-DNS-SVCB-Keys = sf-list
+Proxy-DNS-Request = sf-item
 ~~~
 
-Each list member corresponds to the numeric version of an SvcParamKey.
+The item value is the name of a SVCB DNS query name.  This is the service
+name, possibly prepended with a label ({{SVCB}} Section 2.3).
+For example, "example.com" or "_foo.example.com" or "_8443._bar.api.example.com".
+
+*TODO*: Should this have a trailing "." to be an FQDN?
+
+The item has the following parameters:
+
+* t: The DNS RR type of the DNS query name, which MUST be specified
+  as an sf-integer (so 64 for SVCB and 65 for HTTPS).  If not
+  specified, the value of 65 (HTTPS) is the default.
+
+* wait: The maximum time the proxy should wait before responding to
+  the CONNECT* request while waiting for the SVCB resolution to
+  complete, specified as an sf-integer.  The proxy MAY choose to not
+  wait this long before responding.  If this value is less than 1
+  then proxy SHOULD NOT wait for a SVCB resolution and should
+  only include SVCB information it has cached.
+  If this parameter is not specified, proxies are free to choose
+  a reasonable default.
+
+* params: Clients can optionally limit which SVCB parameters they'd like
+  to receive by optionally specifying this parameter.  Its value MUST
+  be an inner-list whose members are sf-integer items.
+  Each list member corresponds to the numeric version of an SvcParamKey.
+  If not specified, proxies SHOULD return the full set of SVCB parameters
+  for each SVCB RR.
+
+* version: Clients can optionally specify which versions of this specification
+  they support.  Its value MUST be an inner-list whose members are sf-string
+  values.  This current draft specifies version "draft-01".
 
-For example, a client wanting to receive ALPN and ECH Config parameters would
-send a request for 1 (alpn) and 5 (echconfig):
+* u: Clients can indicate whether they want DNS information about
```

Comment:
Why `u`?

Reply:
Ah, it's to get the Used response. Maybe call this out.

Reply:
Should we just call it "used"?  (especially since it is boolean and default-true so not included most of the time?)